TestComponentCSharp/NonAgileClass.cpp

```diff
@@ -0,0 +1,125 @@
+#include "pch.h"
+#include "NonAgileClass.h"
+#include "NonAgileClass.g.cpp"
+
+using namespace winrt;
+
+namespace
+{
+    struct __declspec(uuid("624cd4e1-d007-43b1-9c03-af4d3e6258c4")) __declspec(novtable)
+        INonAgileBindableVectorChangedEventHandler : ::IUnknown
+    {
+        virtual int32_t __stdcall Invoke(void*, void*) noexcept = 0;
+    };
+
+    struct NonAgileDelegate : implements<NonAgileDelegate, non_agile, INonAgileBindableVectorChangedEventHandler, IMarshal>
+    {
+        NonAgileDelegate()
+        {
+        }
+        
+        int32_t __stdcall Invoke(void* p1, void* p2) noexcept override
+        {
+            VectorChanged(*reinterpret_cast<Microsoft::UI::Xaml::Interop::IBindableObservableVector const*>(&p1),
+                          *reinterpret_cast<Windows::Foundation::IInspectable const*>(&p2));
+            return S_OK;
+        }
+
+        HRESULT __stdcall GetUnmarshalClass(REFIID riid, void* pv, DWORD dwDestContext, void* pvDestContext, DWORD mshlflags, CLSID* pCid) noexcept final
+        {
+            if (m_marshaler)

```

Comment:
probably don't need all these if checks, since you're aggregating the FTM unconditionally (and WINRT_VERIFY-ing it)